client/state/themes/actions.js

```diff
@@ -668,3 +672,33 @@ export function deleteTheme( themeId, siteId ) {
 			} );
 	};
 }
+
+/**
+ * Shows dialog asking user to confirm delete of theme
+ * from jetpack site. Deletes theme if user confirms.
+ *
+ * @param {String} themeId -- Theme to delete
+ * @param {Number} siteId -- Site to delete theme from
+ *
+ * @return {Function} Action thunk
+ */
+export function confirmDelete( themeId, siteId ) {
+	return ( dispatch, getState ) => {
+		const { name: themeName } = getTheme( getState(), siteId, themeId );
+		const siteSlug = getSiteSlug( getState(), siteId );
+		accept(
+			i18n.translate(
+				'Are you sure you want to delete %(themeName)s from %(siteSlug)s?',
+				{ args: { themeName, siteSlug }, context: 'Themes: theme delete confirmation dialog' }
+			),
+			( accepted ) => {
+				accepted && dispatch( deleteTheme( themeId, siteId ) );
+			},
+			i18n.translate(
+				'Delete %(themeName)s',
+				{ args: { themeName }, context: 'Themes: theme delete dialog confirm button' }
+			),
+			i18n.translate( 'Back', { context: 'Theme: theme delete dialog back button' } )
```

Comment:
@seear Incorporating this suggestion would have been great. The contexts of the strings that you add in this PR are not very useful and makes it impossible to re-use already existing translations (might have been good as a comment, though!). On https://translate.wordpress.com/2016/11/24/context-vs-comment/ we tried to explain what "Context" actually means for translations. Thanks for your consideration!